Make the demo pretty

schmmd opened the pull request.

schmmd commented:
@joelgrus care to review?  I have one bug in the ReactJS code I would like to resolve before merging however.

matt-gardner commented:
Is this another one that should be in the website branch / repo?

joelgrus commented:
this is the same challenge I had, which is that some of these changes are code changes and some are website changes, so this really needs to be two different PRs, but that's a difficult workflow

matt-gardner commented:
I was more wondering if it's even possible to put this in the website branch, or if it _has_ to live here.  There are references to `./allennlp/service/static/`, for instance.

schmmd commented:
@matt-gardner no--this should not be on the website branch.  The web demo is a python application, which is entirely separate from our website, which is entirely static files.

The web demo is entirely separate from the web site.

schmmd commented:
It is possible to split the files across GitHub pages and here, but that comes with some downsides that I find unacceptable:

* It's harder to run a development instance of the demo.
* The code is split across two branches, yet needs to be in sync.

I don't see real advantages.

matt-gardner commented:
Yeah, I agree.  We've talked about someone being able to spin up an instance of the demo for themselves, and for that to work the code has to all be here.  I was just wondering; thanks for the clarification.

joelgrus commented on allennlp/commands/serve.py:
```diff
@@ -1,13 +1,13 @@
             'serve', description=description, help='Run the web service and demo.')
 
-    subparser.add_argument('--backend', metavar='backend', type=str, choices=['flask', 'sanic'], default='flask',
+    subparser.add_argument('--backend', metavar='backend', type=str, choices=['sanic'], default='sanic',
```
if we're only supporting one backend, we shouldn't even make this an option.

joelgrus commented on allennlp/service/server_sanic.py:
```diff
@@ -6,8 +6,8 @@
-LOGGING['handlers']['accessTimedRotatingFile']['filename'] = '/tmp/sanic_access.log'
-LOGGING['handlers']['errorTimedRotatingFile']['filename'] = '/tmp/sanic_error.log'
+# LOGGING['handlers']['accessTimedRotatingFile']['filename'] = '/tmp/sanic_access.log'
+# LOGGING['handlers']['errorTimedRotatingFile']['filename'] = '/tmp/sanic_error.log'
```
why did you comment these out? they were there because sanic was creating these files in the project root every time you ran the unit tests

schmmd replied:
I received an error which I resolved by commenting these out.

matt-gardner replied:
@joelgrus fixed the issue that made this necessary.  We can just remove these lines.  The error was because you upgraded sanic - this was the reason sanic was pinned in the first place.

schmmd replied:
I will remove those lines.  @matt-gardner you had brought up this cause in the past (upgrading sanic) and I found it to be true.  I don't know how, but I repeatedly bust my conda environment.

matt-gardner replied:
Also, in general, I'm pretty strongly against checking in commented-out code - just remove it if it should be removed (and it should be here).

schmmd replied:
I'll also test to make sure the issue is resolved.

matt-gardner replied:
Well, actually, I think it's resolved with a minor change that I have made locally, and will be submitting a PR for today.  I'll do that now.

joelgrus commented on allennlp/service/server_sanic.py:
```diff
@@ -19,22 +19,22 @@ def run(port: int) -> None:
-        model = app.servables.get(model_name.lower())
-        if model is None:
-            raise ServerError("unknown model: {}".format(model_name), status_code=400)
+      """make a prediction using the specified model and return the results"""
```
why did you change the indentation here?

schmmd commented:
I upgrade sanic and the commented lines to move the log files.  This resolves the log file issues.

schmmd commented:
@joelgrus take another look?

* Tests pass.
* Linting passes.
* I deleted the sanic log comments and upgraded sanic.
* I removed the `--backend` flag.

joelgrus commented on allennlp/commands/serve.py:
```diff
@@ -1,24 +1,17 @@
-        server_sanic.run(args.port)
-    else:
-        raise Exception("Unsupported backend: " + args.backend)
+    server_sanic.run(args.port)
```
should we also add a `num_workers` arg and pass it to `run`?

schmmd replied:
I'll address this in a separate PR.  I'm still not clear whether we should do this or just have more replicas, but it would be good to be ready.
